Source/D3D11VideoSampleProvider.h

```diff
@@ -231,13 +233,11 @@ class D3D11VideoSampleProvider : public UncompressedVideoSampleProvider, public
         {
             if (isCompatible)
             {
-                bool needsReinit = device;
+                bool needsReinit = device.get();
 
                 SAFE_RELEASE(device);
```

Comment:
When switching to com_ptr, we need to remove all manual AddRef and Release calls, otherwise we mess up com_ptr's automatic ref counting. All SAFE_RELEASE calls must be either replaced by NULL assignments, or in this specific case here we can just delete them, because we assign new values immediately, any previous values are automatically unref'd.